src/and-cli-github-issue.ts

```diff
@@ -0,0 +1,252 @@
+import { Echo } from "./modules/echo";
+import { Github } from "./modules/github";
+import program from "commander";
+import { OptionStringBuilder } from "./utilities/option-string-builder";
+import { Issue } from "./interfaces/github/issue";
+import { table } from "table";
+import { CreateIssueDto } from "./interfaces/github/create-issue-dto";
+import { Prompt } from "./modules/prompt";
+import shell from "shelljs";
+import { Formatters } from "./modules/formatters";
+import { CloneIssueSourceDto } from "./interfaces/github/clone-issue-source-dto";
+import { CloneIssueDestinationDto } from "./interfaces/github/clone-issue-destination-dto";
+import { CommandDefinitions } from "./modules/command-definitions";
+
+CommandRunner.run(async () => {
+    // -----------------------------------------------------------------------------------------
+    // #region Constants
+    // -----------------------------------------------------------------------------------------
+
+    const REPO_ISSUE_EXAMPLE = "repo#issue";
+
+    const ADD_OPTION = new OptionStringBuilder("add", "a");
+    const BODY_OPTION = new OptionStringBuilder("body <body>", "b");
+    const CLONE_FROM_OPTION = new OptionStringBuilder(
+        `clone-from <${REPO_ISSUE_EXAMPLE}>`,
+        "c"
+    );
+    const LIST_OPTION = new OptionStringBuilder("list", "l");
+    const REPO_OPTION = new OptionStringBuilder("repo <repo>", "R");
+    const TITLE_OPTION = new OptionStringBuilder("title <title>", "t");
+
+    // #endregion Constants
+
+    // -----------------------------------------------------------------------------------------
+    // #region Functions
+    // -----------------------------------------------------------------------------------------
+
+    const clone = async (cloneFrom: string, destinationRepo?: string) => {
+        validatePatternOrExit(cloneFrom);
+
+        if (StringUtils.isEmpty(destinationRepo)) {
+            destinationRepo = await Prompt.questionAsync(
+                "What repository should this issue be cloned to? "
+            );
+        }
+
+        const [sourceRepo, sourceNumber] = cloneFrom.split("#");
+
+        const sourceIssue: CloneIssueSourceDto = {
+            number: Number(sourceNumber),
+            owner: Github.andcultureOrg,
+            repo: sourceRepo,
+        };
+        const destinationIssue: CloneIssueDestinationDto = {
+            owner: Github.andcultureOrg,
+            repo: destinationRepo,
+        };
+
+        const clonedIssue = await Github.cloneIssueToRepository(
+            sourceIssue,
+            destinationIssue
+        );
+
+        if (clonedIssue == null) {
+            shell.exit(1);
+        }
+
+        printCloneSuccess(sourceIssue, destinationIssue, clonedIssue);
+    };
+
+    const dtoHasRequiredFields = (
+        dto: Partial<CreateIssueDto>
+    ): dto is CreateIssueDto => {
+        const { repo, title, body } = dto;
+        if (
+            StringUtils.isEmpty(repo) ||
```

Comment:
Minor: just for readability maybe you can assign the conditions to a variable and put the variable in the if statement. Like 

`const isEmpty = StringUtils.isEmpty(repot) ||  StringUtils.isEmpty(title) || StringUtils.isEmpty(body)`
`if (isEmpty) {
return false;
}
`

Reply:
True - honestly, might just be more concise to return the result of `StringUtils.hasValue(repo) && StringUtils.hasValue(title) && StringUtils.hasValue(body)` instead of doing a manual true/false return or pulling out a variable for it